tests/e2e/login/privacy-policy.spec.js

```diff
@@ -76,109 +74,122 @@ describe('Privacy policy', () => {
-
-      commonElements.goToLoginPage();
-      loginPage.login('online', password);
-      commonElements.calm(); // no privacy policy on 2nd login
-
-      commonElements.goToLoginPage();
-      loginPage.login('online', password, false, 'fr'); // login in french now
-      const contentFr = element(by.css('#privacy-policy-wrapper .html-content'));
-      expect(helper.getTextFromElement(contentFr)).toEqual('Politique de confidentialité en Francais\nPlus de markup');
-      element(by.css('#privacy-policy-wrapper .btn')).click();
-      commonElements.calm();
+      await acceptButton.click();
+      await commonElements.calmNative();
+      await utils.resetBrowserNative();
+      // After accepting, no privacy policy on next load
+      await commonElements.calmNative();
+      await closeTourModal();
+      await helper.handleUpdateModalNative();
+
+      // Check display when loading privacy policy page
+      await browser.get(utils.getBaseUrl() + 'privacy-policy');
+      const privacyPolicyContainer = element(by.css('.privacy-policy.configuration'));
+      await helper.waitElementToBeVisibleNative(privacyPolicyContainer);
+      content = await helper.getTextFromElementNative(privacyPolicyContainer);
+      expect(content).toEqual('English Privacy Policy\nMore markup');
+
+      // No privacy policy on 2nd login
+      await commonElements.goToLoginPageNative();
+      await loginPage.loginNative('online', password);
+      await commonElements.calmNative();
+
+      // After login in french, check that privacy policy was prompted to user again
+      await commonElements.goToLoginPageNative();
+      await loginPage.loginNative('online', password, false, 'fr');
+      const contentFr = await helper.getTextFromElementNative(element(by.css('#privacy-policy-wrapper .html-content')));
+      expect(contentFr).toEqual('Politique de confidentialité en Francais\nPlus de markup');
+      await element(by.css('#privacy-policy-wrapper .btn')).click();
+      await commonElements.calmNative();
     });
   });
 
   describe('for a french offline user', () => {
-    afterEach(() => utils.deleteUsers([offlineUser]));
+    afterEach(async () => {
+      await utils.deleteUsersNative([offlineUser]);
+    });
 
-    it('should show the correct privacy policy on login', () => {
-      browser.wait(() => utils.createUsers([offlineUser]).then(() => true));
-      commonElements.goToLoginPage();
-      loginPage.login('offline', password, false, 'fr');
+    it('should show the correct privacy policy on login', async () => {
+      await utils.createUsersNative([offlineUser]);
 
-      helper.waitElementToPresent(element(by.css('#privacy-policy-wrapper')));
+      // After first login in french, check that privacy policy was prompted to user
+      await commonElements.goToLoginPageNative();
+      await loginPage.loginNative('offline', password, false, 'fr');
+      await browser.sleep(1000); // overriding default language.
```

Comment:
I think the call to brower.sleep is a bit flakey and the next call to waitElementToPresent might be a bit off since it only checks the element is in the dom. Might use `waitUntilReadyNative` as it checks for the element to be visible before continuing.

Reply:
Removed!